src/main/java/com/hivemq/client/internal/mqtt/message/publish/MqttPublish.java

```diff
@@ -86,6 +88,24 @@ public MqttPublish(
         this.responseTopic = responseTopic;
         this.correlationData = correlationData;
         this.confirmable = confirmable;
+        this.localCorrelationData = localCorrelationData;
+    }
+
+    public MqttPublish(
+            final @NotNull MqttTopicImpl topic,
+            final @Nullable ByteBuffer payload,
+            final @NotNull MqttQos qos,
+            final boolean retain,
+            final long messageExpiryInterval,
+            final @Nullable Mqtt5PayloadFormatIndicator payloadFormatIndicator,
+            final @Nullable MqttUtf8StringImpl contentType,
+            final @Nullable MqttTopicImpl responseTopic,
+            final @Nullable ByteBuffer correlationData,
+            final @NotNull MqttUserPropertiesImpl userProperties,
+            final @Nullable Confirmable confirmable) {
+
+        this(topic, payload, qos, retain, messageExpiryInterval, payloadFormatIndicator, contentType,
+                responseTopic, correlationData, userProperties, confirmable, null);
```

Comment:
Please do not add secondary constructors. We have the builder pattern for a reason. Sadly, you have to touch some tests :(

Reply:
I know, but since the constructor is public and might be used by other users of our client I wanted to retain compatibility.
I am fine removing these if compatibility isn't required.

Reply:
This calls resides in the internal package, so compatibility is not required. Internal classes may change at any time.

Reply:
Yannick is correct, the internal package must not be used:

> All code inside com.hivemq.client.internal packages must not be used directly. It can change at any time and is not part of the public API.

https://github.com/hivemq/hivemq-mqtt-client#versioning

Reply:
Removed it.